README.md

```diff
@@ -26,134 +26,154 @@ composer require opentracing/opentracing
-    GlobalTracer::set(new MyTracer());
+use OpenTracing\GlobalTracer;
+
+GlobalTracer::set(new MyTracerImplementation());
 ```
 
 ### Creating a Span given an existing Request
 
-To start a new `Span`, you can use the `StartSpanFromContext` method.
+To start a new `Span`, you can use the `startActiveSpan` method.
 
 ```php
-    use OpenTracing\Formats;
-    use OpenTracing\GlobalTracer;
+use OpenTracing\Formats;
+use OpenTracing\GlobalTracer;
 
-    ...
+...
 
-    $spanContext = GlobalTracer::get()->extract(
-        Formats\HTTP_HEADERS,
-        $request->getHeaders()
-    );
+$spanContext = GlobalTracer::get()->extract(
+    Formats\HTTP_HEADERS,
+    getallheaders()
+);
+
+function doSomething(SpanContext $spanContext, ...) {
+    ...
     
-    function doSomething(SpanContext $spanContext, ...) {
-        ...
-        
-        $span = GlobalTracer::get()->startSpan('my_span', [
-        	'child_of' => $spanContext,
-        ]);
-        
-        ...
-        
-        $span->log([
-            'event' => 'soft error',
-            'type' => 'cache timeout',
-            'waiter.millis' => 1500,
-        ])
-        
-        $span->finish();
-    }
+    $span = GlobalTracer::get()->startManualSpan('my_span', ['child_of' => $spanContext]);
+    
+    ...
+    
+    $span->log([
+        'event' => 'soft error',
+        'type' => 'cache timeout',
+        'waiter.millis' => 1500,
+    ])
+    
+    $span->finish();
+}
 ```
 
 ### Starting an empty trace by creating a "root span"
 
-It's always possible to create a "root" `Span` with no parent or other causal
-reference.
+It's always possible to create a "root" `Span` with no parent or other causal reference.
 
 ```php
-    $span = $tracer->startSpan('my_span');
-    ...
-    $span->finish();
+$span = $tracer->startActiveSpan('my_first_span');
+...
+$span->finish();
 ```
 
-### Creating a (child) Span given an existing (parent) Span
+#### Creating a child span assigning parent manually
 
 ```php
-    function xyz(Span $parentSpan, ...) {
-        ...
-        $span = GlobalTracer::get()->startSpan(
-        	'my_span',
-        	[
-        		'child_of' => $parentSpan,
-        	]
-        );
-        
-        $span->finish();
-        ...
-    }
+$parent = GlobalTracer::get()->startManualSpan('parent');
+
+$child = GlobalTracer::get()->startManualSpan('child', [
+    'child_of' => $parent
+]);
+
+...
+
+$child->finish();
+
+...
+
+$parent->finish();
 ```
 
-### Serializing to the wire
+#### Creating a child span using automatic active span management
+
+Every new span will take the active span as parent and it will take its spot.
 
 ```php
-    use OpenTracing\GlobalTracer;
-    use OpenTracing\Formats;
-    
+	$parent = GlobalTracer::get()->startActiveSpan('parent');        
+	
+	...
+
+    // Since the parent span has been created by using startActiveSpan we don't need
+    // to pass a reference for this child span
+    $child = GlobalTracer::get()->startActiveSpan('my_second_span');
```

Comment:
it's worth mentioning that the active parent should be inherited regardless of whether the child span is started as manual or active.